flutter/lib/src/file_system_transport.dart

```diff
@@ -29,4 +33,9 @@ class FileSystemTransport implements Transport {
 
     return envelope.header.eventId;
   }
+
+  @override
+  void recordLostEvent(DiscardReason reason, DataCategory category) {
+    _clientReportRecorder.recordLostEvent(reason, category);
+  }
```

Comment:
l: HttpTransport has the same implementation. Can we use mixin somehow and avoid duplicating code and reuse it in future transporte classes or client transport implementation?

Reply:
Looked at this and there are some issues i encountered:

-  Both implementations (Http/File) live in different SDKs and we'd have to provide some public component that they can share.
- We'd potentially run in issues with unit testing as mocking will be more difficult with the Mixin.
- If we do the mixin on the `Transport` interface, we don't have access to the recorder, as it is a detail of the implementations.

Don't think it's worth the effort, as we are only calling a method and there's not much to share, or am i missing something?

Reply:
Not relevant anymore, as the recorder instance moved out of the transport classes with the last refactoring.